src/schedule/SchedulePing.ts

```diff
@@ -30,33 +30,42 @@ export class SchedulePing {
     try {
       await this.checkActiveSchedule();
     } catch (e) {
-      this.logger.error(errorMessage, MomoErrorType.internal, {}, e);
+      this.logger.error(errorMessage, MomoErrorType.internal, this.schedulesRepository.getLogData(), e);
     }
     this.handle = setSafeInterval(this.checkActiveSchedule.bind(this), this.interval, this.logger, errorMessage);
   }
 
   private async checkActiveSchedule(): Promise<void> {
-    const active = await this.schedulesRepository.isActiveSchedule(this.scheduleName);
-    this.logger.debug(`This schedule is ${active ? '' : 'not '}active`);
-    if (active) {
-      await this.schedulesRepository.ping(this.scheduleId);
-      if (this.startJobsStatus === StartJobsStatus.notStarted) {
-        this.startJobsStatus = StartJobsStatus.inProgress;
-        this.logger.debug('This schedule just turned active');
-
-        await this.startAllJobs();
-
-        this.startJobsStatus = StartJobsStatus.finished;
-        this.logger.debug('Finished starting scheduled jobs');
-      }
+    const now = DateTime.now().toMillis();
+    const active = await this.schedulesRepository.isActiveSchedule(now);
+    if (!active) {
+      this.logger.debug('This schedule is not active', this.schedulesRepository.getLogData());
+      return;
+    }
+
+    if (!(await this.schedulesRepository.setActiveSchedule(now))) {
```

Comment:
Can we extract the condition to a variable for documentation purposes?
```suggestion
    const somethingThatDescribesWhatIsReturned = await this.schedulesRepository.setActiveSchedule(now)
    if (!somethingThatDescribesWhatIsReturned) {
```

Reply:
I refactored this again, now calling the "isActiveSchedule" inside the "setActiveSchedule" in SchedulesRepository, making the code in SchedulePing clearer.